File: doc-examples/jdbc-example-java/src/test/java/example/ProductManagerSpec.java
package example;

import jakarta.inject.Inject;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

public abstract class ProductManagerSpec {

    @Inject
    ProductManager productManager;

    @Inject
    ProductRepository productRepository;

    @Inject
    ManufacturerRepository manufacturerRepository;

    @BeforeAll
    void setupTest() {
        if(productRepository != null) {
            productRepository.deleteAll();
        }
        if(manufacturerRepository != null) {
            manufacturerRepository.deleteAll();
        }
    }

    @Test
    void testProductManager() {
        Manufacturer apple = manufacturerRepository.save("Apple");
        productManager.save("VR", apple);

        Product product = productManager.find("VR");
        assertEquals("VR", product.getName());
    }

    @Test
    void testProductManagerUsingRepo() {
        Manufacturer intel = manufacturerRepository.save("Intel");
        productManager.saveUsingRepo("Processor", intel);

        Product product = productManager.findUsingRepo("Processor");
        assertEquals("Processor", product.getName());

        product = productManager.findUsingRepo("NonExistingProduct");
        assertNull(product);
    }

}

File: doc-examples/jdbc-example-python/src/test/python/example/ProductManagerSpec.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import BeforeEach, Test

from example.ManufacturerRepository import ManufacturerRepository
from example.ProductManager import ProductManager
from example.ProductRepository import ProductRepository


@MicronautTest
class ProductManagerSpec:

    productManager: Annotated[ProductManager, Inject]
    productRepository: Annotated[ProductRepository, Inject]
    manufacturerRepository: Annotated[ManufacturerRepository, Inject]

    @BeforeEach
    def setupTest(self):
        self.productRepository.deleteAll()
        self.manufacturerRepository.deleteAll()

    @Test
    def testProductManager(self):
        apple = self.manufacturerRepository.save("Apple")
        self.productManager.save("VR", apple)
        product = self.productManager.find("VR")
        assert product.name == "VR"

    @Test
    def testProductManagerUsingRepo(self):
        intel = self.manufacturerRepository.save("Intel")
        self.productManager.save_using_repo("Processor", intel)
        product = self.productManager.find_using_repo("Processor")
        assert product.name == "Processor"

        product = self.productManager.find_using_repo("NonExistingProduct")
        assert product is None
